### PR TITLE
Add automated Schema File publish action

### PR DESCRIPTION
**Pull request by tigrannajaryan**
The action should be triggered manually after every spec release
and will copy/update schema files from this repository to the
website repository so that they become available publicly.

The publish-schemas action after manual triggering will create a PR
in https://github.com/open-telemetry/opentelemetry.io repo, and the
PR needs to be approved and merged manually.

This is similar to how we publish doc updates from language repos
to the website repo.

Resolves https://github.com/open-telemetry/opentelemetry.io/issues/577
Resolves https://github.com/open-telemetry/opentelemetry-specification/issues/1693

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@open-telemetry/docs-approvers please review.

File: .github/workflows/publish-schemas.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+          API_TOKEN_GITHUB: ${{secrets.DOC_UPDATE_TOKEN}}
+          # Destination repo should always be 'open-telemetry/opentelemetry.io'
+          DESTINATION_REPO: open-telemetry/opentelemetry.io
+          # Destination path should be the absolute path to directory to publish in
+          DESTINATION_PATH: static/schemas
+          # Source path should be 'schemas', all files and folders are copied from here to dest
+          SOURCE_PATH: schemas
+        run: |
+          TARGET_DIR=$(mktemp -d)
+          export GITHUB_TOKEN=$API_TOKEN_GITHUB
```
Comment (SergeyKanzhelev) [inline .github/workflows/publish-schemas.yml:24]: @austinlparker are you ok with the token reuse here?

File: .github/workflows/publish-schemas.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+          git config --global user.name austinlparker
+          git config --global user.email austin@lightstep.com
+          git clone "https://$API_TOKEN_GITHUB@github.com/$DESTINATION_REPO.git" "$TARGET_DIR"
+          rsync -av --delete "$SOURCE_PATH/" "$TARGET_DIR/$DESTINATION_PATH/"
+          cd "$TARGET_DIR"
+          git checkout -b docs-$GITHUB_REPOSITORY-$GITHUB_SHA
```
Comment (SergeyKanzhelev) [inline .github/workflows/publish-schemas.yml:30]: `GITHUB_SHA` this is SHA of specs repo, not protos, correct?

Comment (tigrannajaryan) [reply .github/workflows/publish-schemas.yml]: Yes, I think so. (This is code is copied from Java repo, I didn't write it, I believe is @austinlparker the author).

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
@austinlparker Is on paternity leave, so we need to proceed without him. Anybody we could poke for the time being? Maybe @tedsuo ? Else let's just go ahead and merge it.

**Review by mtwo — APPROVED** [review]

**Comment by carlosalberto** [comment]
@SergeyKanzhelev @tigrannajaryan Let's merge this? I think it deserves a run ;)

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
Assuming Austin is fine with the token reuse.

**Pull request merged**